scripts/asm-installer/install_asm

```diff
@@ -728,6 +733,47 @@ validate_istio_version() {
   fi
 }
 
+validate_asm_version() {
+  info "Checking existing ASM version(s)..."
+  local OUTPUT
+  OUTPUT="$(retry 3 ./istio-${RELEASE}/bin/istioctl version -o json)"
+  local FOUND_INVALID_VERSION
+  FOUND_INVALID_VERSION="false"
+  for version in $(echo "${OUTPUT}" | jq -r '.meshVersion[].Info.version'); do
+    if ! [[ "$version" =~ "asm" ]]; then
+      fatal "Cannot upgrade from version $version. Only upgrade from ASM distribution is supported."
+    fi
+
+    # if asm version found, pattern: 1.6.11-asm.1-586f900508ad482ed32b830dd15f6c54b32b93ed
+    local VERSION_MAJOR VERSION_MINOR VERSION_POINT VERSION_REV
+    IFS="." read VERSION_MAJOR VERSION_MINOR VERSION_POINT VERSION_REV <<< "$version"
+    VERSION_POINT="$(echo "$VERSION_POINT" | sed 's/-.*//')"
+    VERSION_REV="$(echo "$VERSION_REV" | sed 's/-.*//')"
+    if [[ "$VERSION_MAJOR" -ne 1 || "$VERSION_MINOR" -lt 6 || "$VERSION_MINOR" -ge "$MINOR" || \
+      ( "$VERSION_MINOR" -eq "$MINOR" && \
+        ( "$VERSION_POINT" -ge "$POINT" || \
+          ("$VERSION_POINT" -eq "$POINT" && "$VERSION_REV" -ge "$REV"))) ]]; then
+      info "  $version (not suitable for migration)"
+      FOUND_INVALID_VERSION="true"
+    else
+      info "  $version (suitable for migration)"
+    fi
+  done
+  if [[ "$FOUND_INVALID_VERSION" = "true" ]]; then
+    fatal "Upgrade requires all existing control planes to be between versions 1.6.0-asm.0 (inclusive) and ${RELEASE} (exclusive)."
+  fi
+}
+
+validate_ca_consistency() {
+  info "Checking existing CA..."
+  local INSTALLED_CA
+  INSTALLED_CA="$(kubectl -n istio-system get pod -l istio=ingressgateway -o jsonpath='{.items[].spec.containers[].env[?(@.name=="CA_ADDR")].value}')"
+  if [[ ("${CA}" = "citadel" && ! "${INSTALLED_CA}" =~ istiod.*istio\-system\.svc) || \
```

Comment:
Not sure if we need to handle it this in this commit, but do we know this will match even w/custom certs in Citadel?

Reply:
I think we cannot yet support customer CA with scripts due to additional complexities so this will just fail fast right?

Reply:
Someone might have installed manually and are trying to use the script to upgrade.

Reply:
updating this to pivot on meshCA check only

Reply:
```
if [[ ("${INSTALLED_CA}" =~ meshca\.googleapis\.com && ! "${CA}" = "mesh_ca") || \
        ( ! "${INSTALLED_CA}" =~ meshca\.googleapis\.com && ! "${CA}" = "citadel") ]]; then
    fatal "CA cannot be switched while performing upgrade. Please use ${CA}."
fi
```

is this what you were thinking?

Reply:
Maybe a bit easier to read:

```
is_meshca() {
  return [[ "${INSTALLED_CA}" =~ meshca\.googleapis\.com ]]
}

if is_meshca && "${CA}" == "citadel" || ! is_meshca && "${CA}" == "mesh_ca"; then
```

Reply:
agreed. I wonder if we need the condition to be grouped otherwise `is_meshca && ... || ! is_meshca && ..` will always be true? I'm not sure what's the posh way to group conditions..

Reply:
using `[[ ]]` for grouping